docs/syntax/pattern.md

```diff
@@ -256,6 +256,22 @@ c = ...
     _        => "none of x and y equal to c"
 end
 ```
+Reference Patterns are useful, for example, when it's necessary to match on the values of numeric variables, but not the type:
+
+```julia
+c = Int16(10) # c is of type Int16
+
+@match c begin
+    10.0 => "there is a match" #pattern is a Float
+    _    => "there is not a match"
+end # => "there is not a match"
+
+@match c begin
+    &10.0 => "there is a match"
+    _    => "there is not a match"
+end # => "there is a match"
+```
+Internally, literal pattern matching behaves with strict equality, similar to the [`===`](https://docs.julialang.org/en/v1/base/base/#Core.:===) operator in base Julia. Reference patterns behave more like the [`==`](https://docs.julialang.org/en/v1/base/math/#Base.:==) operator in base Julia, where the type of the numeric variable is ignored, and only abstract values are compared.   
```

Comment:
Now a literal pattern produces `===` only if the literal's type satisfies

https://github.com/thautwarm/MLStyle.jl/blob/591bcceb1e673c1caf7b7419df31208b95e82bf3/src/AbstractPatterns/impl/RedyFlavoured.jl#L215